lib/pyszn/parser.py

```diff
@@ -151,122 +170,93 @@ def parse_txtmeta(txtmeta):
 
-            # Parse line and optional attributes
-            parsed = statement.parseString(line)
-            attrs = process_attributes(parsed)
-            log.debug(parsed.dump())
-            log.debug(attrs)
-
-            # Process link lines
-            if 'link' in parsed:
-
-                link = parsed.link
-                data['links'].append({
-                    'endpoints': (
-                        (link.endpoint_a.node, link.endpoint_a.port),
-                        (link.endpoint_b.node, link.endpoint_b.port),
-                    ),
-                    'attributes': attrs,
-                })
-
-                data['nodes'].append({
-                    'nodes': [link.endpoint_a.node, link.endpoint_b.node],
-                    'attributes': OrderedDict(),
-                })
-                data['ports'].append({
-                    'ports': [
-                        (link.endpoint_a.node, link.endpoint_a.port),
-                        (link.endpoint_b.node, link.endpoint_b.port)
-                    ],
-                    'attributes': OrderedDict(),
-                })
-                continue
-
-            # Process port lines
-            if 'ports' in parsed:
-                data['ports'].append({
-                    'ports': [
-                        (port.node, port.port) for port in parsed.ports
-                    ],
-                    'attributes': attrs,
-                })
-                data['nodes'].append({
-                    'nodes': [port.node for port in parsed.ports],
-                    'attributes': OrderedDict(),
-                })
-                continue
-
-            # Process port lines
-            if 'nodes' in parsed:
-                data['nodes'].append({
-                    'nodes': [node.node for node in parsed.nodes],
-                    'attributes': attrs,
-                })
-                continue
-
-            # Process environment line
-            if 'environment' in parsed:
-                if len(data['environment']) == 0:
-                    data['environment'].update(attrs)
-                else:
-                    raise Exception(
-                        'Multiple declaration of environment attributes: '
-                        '{}'.format(attrs)
-                    )
-                continue
-
-            raise Exception('Unknown line type parsed.')
-
-        except Exception as e:
-            raise ParseException(lineno, raw_line, format_exc()) from e
+    # Process environment line
+    if 'env_spec' in parsed_result:
+        if len(parsed_result['env_spec']) > 1:
+            raise Exception(
+                'Multiple declaration of environment attributes: '
+                '{}'.format(parsed_result['env_spec'][1])
+            )
+        for attr in parsed_result['env_spec'][0]:
+            data['environment'][attr.key] = attr.value
+
+    # Process the links
+    if 'link_spec' in parsed_result:
```

Comment:
link_spec is not a seq, it is a pyparsing `ParsingResult` and so is parsed_result. As such there is no way to assign it to an empty seq or an empty `ParsingResult` since there is no such thing.